youtube_dl/extractor/openload.py

```diff
@@ -302,14 +302,18 @@ def _extract_urls(webpage):
     def _real_extract(self, url):
         video_id = self._match_id(url)
         url = 'https://openload.co/embed/%s/' % video_id
+        url2 = 'https://openload.co/f/%s/' % video_id
         headers = {
             'User-Agent': self._USER_AGENT,
         }
 
         webpage = self._download_webpage(url, video_id, headers=headers)
 
         if 'File not found' in webpage or 'deleted by the owner' in webpage:
-            raise ExtractorError('File not found', expected=True, video_id=video_id)
+            url = url2
+            webpage = self._download_webpage(url, video_id, headers=headers)
+            if 'File not found' in webpage or 'deleted by the owner' in webpage:
+                raise ExtractorError('File not found', expected=True, video_id=video_id)
```

Comment:
Code duplication.

Reply:
The code is duplicated because you need to attempt to download the webpage twice with two different URLs. You will only know to attempt the second URL if the first download attempt fails. If this is unacceptable do you have any suggestions to make the code smell less?

Reply:
That's not an excuse. Code duplication must be avoided by generalizing common pieces of code.

Reply:
@zachee @dstftw 

I had a go at removing the duplication. 

Is this any good?

**Function**

```
def _get_first_valid_downloaded_webpage(urls, video_id, headers):
    for url in urls:
        webpage = self._download_webpage(url, video_id, headers=headers)
        if not ('File not found' in webpage or 'deleted by the owner' in webpage):
            return webpage
    raise ExtractorError('File not found', expected=True, video_id=video_id)
```

**Usage**

```
def _real_extract(self, url):
         video_id = self._match_id(url)
         url = 'https://openload.co/embed/%s/' % video_id
         url2 = 'https://openload.co/f/%s/' % video_id
         headers = {
             'User-Agent': self._USER_AGENT,
         }
         webpage = self._get_first_valid_downloaded_webpage([url, url2], video_id, headers)
```

Reply:
@cacdpa: That's better. Just don't use the name ```url``` - it's already used as a parameter of ```_real_extract```.

Reply:
@cacdpa looks good to me, sorry I didn't get a chance to fix it myself